opencog/query/Implicator.cc

```diff
@@ -49,8 +49,12 @@ bool Implicator::grounding(const HandleMap &var_soln,
 	if (_result_set.size() >= max_results)
 		return true;
 
-	Handle h = inst.instantiate(implicand, var_soln);
-	insert_result(h);
+	// By wrapping this in a try/catch we can ignore the cases when
+	// executions create ill-formed atoms.
+	try {
+		Handle h = inst.instantiate(implicand, var_soln);
+		insert_result(h);
+	} catch(const InvalidParamException& e) {}
```

Comment:
Indeed. I put that specific exception because it is currently the only one bothering me, my plan was to extend as needed. BTW, this code has been tested against backward chainer inferences and has been proven correct for now. That is said if you consider that it is too permissive, I can easily move this to the callback.